src/editor-service/editor-server.ts

```diff
@@ -0,0 +1,167 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import * as util from 'util';
+
+import {FSUrlLoader} from '../url-loader/fs-url-loader';
+import {PackageUrlResolver} from '../url-loader/package-url-resolver';
+
+import {SourcePosition} from './editor-service';
+import {LocalEditorService} from './local-editor-service';
+
+
+/**
+ * The remote editor service protocol is request/response based.
+ * Requests and responses are JSON serialized objects.
+ *
+ * Every request must have a unique identifier. For every request there
+ * will be at most one response with the corresponding identifier. Responses
+ * may come in any order.
+ *
+ * Responses are modeled on settled Promise values. A response is either
+ * successful and resolved or unsuccessful and rejected.
+ *
+ * The types of requests and responses obey the EditorService interface, with
+ * some modification of method calls for the JSON format. See the Request
+ * type for more information.
+ */
+export interface RequestWrapper {
+  id: number;
+  value: Request;
+}
+export interface ResponseWrapper {
+  id: number;
+  value: SettledValue;
+}
+export type SettledValue = Resolution | Rejection;
+export interface Resolution {
+  kind: 'resolution';
+  resolution: any;
+}
+export interface Rejection {
+  kind: 'rejection';
+  rejection: string;
+}
+
+
+export type Request = InitRequest | FileChangedRequest | GetWarningsRequest |
+    GetDocumentationRequest | GetDefinitionRequest |
+    GetTypeaheadCompletionsRequest | ClearCachesRequest;
+export interface InitRequest {
```

Comment:
COnsider moving these interfaces to a file that's imported by both editor-server and remote-editor-service